src/components/api.js

```diff
@@ -0,0 +1,132 @@
+async function getActors() {
+  try {
+    const response = await fetch('/api/actors')
+
+    if (response.ok) {
+      return await response.json()
+    } else {
+      console.log('Server Error:', response.body);
+    }
+  } catch (error) {
+    console.error('Network Error:', error);
+  }
+}
+
+async function getScript(scriptId) {
```

Comment:
In regards to https://github.com/rcheman/PlayReadingParty/pull/49/files#r1181779318  
we have a bunch of different return styles here:
* return json on success or nothing (getScript, getActors, getCurrentActorCharacters, getCharacters)
* return json on success or error string (getScriptTitles, newActor)
* return bool on success and failure or nothing if a network error (deleteActor, postPosition)
* return response object on success and failure or nothing if a network error (deleteScript)
* return response object in all cases (uploadScript)
  
We should standardize on one or a small number of styles that are consistent and that covers all error cases.  
As an example, we could return json in all cases by returning data in one specific format:
```js
// for success
return {
    success: true,
    data: await response.json()
};

// for failure
return {
    success: false,
    data: error.message
}

// for network failure
return {
    success: false,
    data: `Network failure when performing ${endpoint} api call. Context: ${extraInformativeData}`
};
```
This would allow callers to always rely on the success field. For any api function, there would be valid data if success were true. 
Later when you or someone else is looking at a usage of an api call, you can easily see if errors are being handled properly. If the code is checking the success field, it's good. We don't need to look at the specific implementation of the api function to know whether we're using it incorrectly.  
  
When all usages look like this:
```js
const result = await getActors();
if (result.success) {
    setActors(result.data);
} else {
    alert("OMG we couldn't do the thing you wanted due to this error: " + result.data);
}
```
then this stands out as suspicious:
```js
setCharacters(await getCharacters());
```

Reply:
This is extremely helpful feedback. Per our discussion, I had a feeling that returning a variety of types wasn't the cleanest way of doing things, but wasn't quite sure how to go about resolving that.